bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/ChannelProviderWrapper.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.config;
+
+import com.google.api.gax.grpc.InstantiatingGrpcChannelProvider;
+import com.google.api.gax.rpc.TransportChannel;
+import com.google.api.gax.rpc.TransportChannelProvider;
+import com.google.auth.Credentials;
+import com.google.cloud.bigtable.metrics.BigtableClientMetrics;
+import com.google.cloud.bigtable.metrics.Counter;
+import com.google.cloud.bigtable.metrics.Meter;
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+
+// TODO: Need to figure out how this will connect with grpc Channel pool to get the count
+public class ChannelProviderWrapper implements TransportChannelProvider {
```

Comment:
I think it will be cleaner to implement this as a ClientInterceptor and install it using InstantiatingGrpcChannelProvider.Builder#setInterceptorProvider()